packages/celotool/src/lib/geth.ts

```diff
@@ -614,3 +639,501 @@ export const transferERC20Token = async (
+
+  if (verbose) {
+    console.log(`wrote   genesis to ${genesisPath}`)
+  }
+}
+
+export async function snapshotDatadir(
+  runPath: string,
+  instance: GethInstanceConfig,
+  verbose: boolean
+) {
+  if (verbose) {
+    console.log('snapshotting data dir')
+  }
+
+  // Sometimes the socket is still present, preventing us from snapshotting.
+  await spawnCmd('rm', [`${getDatadir(runPath, instance)}/geth.ipc`], { silent: true })
+  await spawnCmdWithExitOnFailure('cp', [
+    '-r',
+    getDatadir(runPath, instance),
+    getSnapshotdir(runPath, instance),
+  ])
+}
+
+export async function restoreDatadir(runPath: string, instance: GethInstanceConfig) {
+  const datadir = getDatadir(runPath, instance)
+  const snapshotdir = getSnapshotdir(runPath, instance)
+
+  console.info(`geth:${instance.name}: restore datadir: ${datadir}`)
+
+  await spawnCmdWithExitOnFailure('rm', ['-rf', datadir], { silent: true })
+  await spawnCmdWithExitOnFailure('cp', ['-r', snapshotdir, datadir], { silent: true })
+}
+
+export async function buildGeth(gethPath: string) {
+  await spawnCmdWithExitOnFailure('make', ['geth'], { cwd: gethPath })
+}
+
+export async function resetDataDir(dataDir: string, verbose: boolean) {
+  await spawnCmd('rm', ['-rf', dataDir], { silent: !verbose })
+  await spawnCmd('mkdir', [dataDir], { silent: !verbose })
+}
+
+export async function checkoutGethRepo(branch: string, gethPath: string) {
+  await spawnCmdWithExitOnFailure('rm', ['-rf', gethPath])
+  await spawnCmdWithExitOnFailure('git', [
+    'clone',
+    '--depth',
+    '1',
+    'https://github.com/celo-org/celo-blockchain.git',
+    gethPath,
+    '-b',
+    branch,
+  ])
+  await spawnCmdWithExitOnFailure('git', ['checkout', branch], { cwd: gethPath })
+}
+
+export function spawnWithLog(cmd: string, args: string[], logsFilepath: string, verbose: boolean) {
+  try {
+    fs.unlinkSync(logsFilepath)
+  } catch (error) {
+    // nothing to do
+  }
+
+  const logStream = fs.createWriteStream(logsFilepath, { flags: 'a' })
+
+  if (verbose) {
+    console.log(cmd, ...args)
+  }
+
+  const p = spawn(cmd, args)
+
+  p.stdout.pipe(logStream)
+  p.stderr.pipe(logStream)
+
+  if (verbose) {
+    p.stdout.pipe(process.stdout)
+    p.stderr.pipe(process.stderr)
+  }
+
+  return p
+}
+
+export async function connectPeers(instances: GethInstanceConfig[]) {
+  const admins = instances.map(({ wsport, rpcport }) => {
+    const url = `${wsport ? 'ws' : 'http'}://localhost:${wsport || rpcport}`
+    return new Admin(url)
+  })
+
+  await Promise.all(
+    admins.map(async (admin, i) => {
+      const enodes = await Promise.all(admins.map(async (a) => (await a.getNodeInfo()).enode))
+      await Promise.all(
+        enodes.map(async (enode, j) => {
+          if (i === j) {
+            return
+          }
+          /*
+          console.log(
+            `connecting ${instances[i].name} with ${instances[j].name} using enode ${enode}`
+          )
+          */
```

Comment:
Please uncomment this as `verbose` logging or delete it. We just don't want to leave commented out code lying around.

Reply:
wrapped in if verbose